source/DSCResources/DSC_CMClientStatusSettings/DSC_CMClientStatusSettings.psm1

```diff
@@ -0,0 +1,262 @@
+        StatusMessageDays      = $statusSettings.StatusInactiveInterval
+        HistoryCleanupDays     = $statusSettings.CleanUpInterval
+    }
+}
+
+<#
+    .SYNOPSIS
+        This will set the desired state.
+
+    .PARAMETER SiteCode
+        Specifies the site code for Configuration Manager site.
+
+    .PARAMETER IsSingleInstance
+        Specifies the resource is a single instance, the value must be 'Yes'.
+
+    .PARAMETER ClientPolicyDays
+        Specifies the data collection intervals for client policy client monitoring activities.
+
+    .PARAMETER HeartbeatDiscoveryDays
+        Specifies the data collection intervals for heartbeat discovery client monitoring activities.
+
+    .PARAMETER SoftwareInventoryDays
+        Specifies the data collection intervals for software inventory client monitoring activities.
+
+    .PARAMETER HardwareInventoryDays
+        Specifies the data collection intervals for hardware inventory client monitoring activities.
+
+    .PARAMETER StatusMessageDays
+        Specifies the data collection intervals for status message client monitoring activities.
+
+    .PARAMETER HistoryCleanupDays
+        Specifies the data collection intervals for status history cleanup client monitoring activities.
+#>
+function Set-TargetResource
+{
+    [CmdletBinding()]
+    param
+    (
+        [Parameter(Mandatory = $true)]
+        [String]
+        $SiteCode,
+
+        [Parameter(Mandatory = $true)]
+        [ValidateSet('Yes')]
+        [String]
+        $IsSingleInstance,
+
+        [Parameter()]
+        [ValidateRange(1,30)]
+        [UInt32]
+        $ClientPolicyDays,
+
+        [Parameter()]
+        [ValidateRange(1,30)]
+        [UInt32]
+        $HeartbeatDiscoveryDays,
+
+        [Parameter()]
+        [ValidateRange(1,30)]
+        [UInt32]
+        $SoftwareInventoryDays,
+
+        [Parameter()]
+        [ValidateRange(1,30)]
+        [UInt32]
+        $HardwareInventoryDays,
+
+        [Parameter()]
+        [ValidateRange(1,30)]
+        [UInt32]
+        $StatusMessageDays,
+
+        [Parameter()]
+        [ValidateRange(0,90)]
+        [UInt32]
+        $HistoryCleanupDays
+    )
+
+    Import-ConfigMgrPowerShellModule -SiteCode $SiteCode
+    Set-Location -Path "$($SiteCode):\"
+    $currentState = Get-TargetResource -SiteCode $SiteCode -IsSingleInstance Yes
+
+    try
+    {
+        foreach ($property in $PSBoundParameters.GetEnumerator())
+        {
+            if ((-not [string]::IsNullOrEmpty($property.Value)) -and ($property.key -ne 'Verbose'))
+            {
+                $currentValue = $currentState.("$($property.Key)")
+
+                if ($property.Value -ne $currentValue)
+                {
+                    Write-Verbose -Message ($script:localizedData.ModifySetting -f $property.Key, $property.Value, $currentValue)
+
+                    $buildingParmas += @{
+                        $($property.Key) = $($property.Value)
```

Comment:
I believe this $($property.Key) = $($property.Value) can be $property.Key = $property.Value